pkg/apis/serving/v1beta1/inference_service_defaults.go

```diff
@@ -58,21 +61,31 @@ func setResourceRequirementDefaults(requirements *v1.ResourceRequirements) {
 }
 
 func (isvc *InferenceService) Default() {
-	logger.Info("Defaulting InferenceService", "namespace", isvc.Namespace, "name", isvc.Name)
-	configMap, err := NewInferenceServicesConfig()
+	mutatorLogger.Info("Defaulting InferenceService", "namespace", isvc.Namespace, "isvc", isvc.Spec.Predictor)
+	cli, err := client.New(config.GetConfigOrDie(), client.Options{})
+	if err != nil {
+		panic("Failed to create client in defaulter")
+	}
+	configMap, err := NewInferenceServicesConfig(cli)
 	if err != nil {
 		panic(err)
 	}
-	isvc.defaultInferenceService(configMap)
+	isvc.DefaultInferenceService(configMap)
 }
 
-func (isvc *InferenceService) defaultInferenceService(config *InferenceServicesConfig) {
+func (isvc *InferenceService) DefaultInferenceService(config *InferenceServicesConfig) {
 	for _, component := range []Component{
 		&isvc.Spec.Predictor,
 		isvc.Spec.Transformer,
 		isvc.Spec.Explainer,
 	} {
-		component.GetImplementation().Default(config)
-		component.GetExtensions().Default(config)
+		if component != nil && !reflect.ValueOf(component).IsNil() {
```

Comment:
Don't reimplement this logic. I'd just call isvc.validateExactlyOne() if you need to.

Reply:
In fact, even if there are multiple specs, GetImplementation will just return the first one. It's okay if they're not all defaulted since validation will fail on the first.

Reply:
good point, fixed